apps/DisplayCluster/resources/ContentWindow.qml

```diff
@@ -27,21 +27,42 @@ Rectangle {
         target: contentwindow; property: "height"; value: height
     }
 
+    function closeWindow() {
+        displaygroup.removeContentWindow(contentwindow.id);
+    }
+
     ContentWindowTouchArea {
         objectName: "ContentWindowTouchArea"
-        id: contentWindowTouchArea
         anchors.fill: parent
 
-        onMoveToFront: controlsFadeAnimation.restart()
+        onActivated: {
+            displaygroup.moveContentWindowToFront(contentwindow.id);
+            controlsFadeAnimation.restart();
+        }
 
         WindowControls {
-            anchors.horizontalCenter: parent.horizontalCenter
-            anchors.bottom: parent.bottom
-            anchors.bottomMargin: 30
-            onClose: contentWindowTouchArea.close()
-            onToggleFullscreen: controller.toggleFullscreen()
             opacity: 0.1
             visible: contentwindow.controlsOpacity > 0 && contentwindow.label != "Dock"
+
+            // controller is not set yet at Component.onCompleted, so
```

Comment:
Shouldn't this be fixed then? The function now compensates for that fact? Will be the same for similar new features we add I guess